python/mxnet/ndarray/ndarray.py

```diff
@@ -1858,10 +2174,8 @@ def full(shape, val, ctx=None, dtype=mx_real_t, out=None):
     >>> mx.nd.full((1, 2), 2.0, dtype='float16').asnumpy()
     array([[ 2.,  2.]], dtype=float16)
     """
-    if ctx is None:
-        ctx = Context.default_ctx
-    dtype = mx_real_t if dtype is None else dtype
-    out = _internal._full(shape=shape, ctx=ctx, dtype=dtype, value=val, out=out)
+    out = empty(shape, ctx, dtype) if out is None else out
```

Comment:
Why revert?

Reply:
Don't remember exactly. @piiswrong once asked me to change something back, not sure if it's this one. But one benefit of changing it is allowing `val` to be any data types that could be assigned to an `NDArray`. Behavior-wise, it's same as before when val is a scalar. Any problem with this change?

Reply:
@piiswrong do you recall?

Reply:
@cjolivier01 here is the link to the conversation.
https://github.com/apache/incubator-mxnet/pull/8340/files/3f88ddf475a5f9920b7e11b606579cc8fc509edb#diff-151ed3672c6bd71b358e13771357696dR1860

Reply:
ok thanks for digging that up!